mythic-melodic/backend
# Playlist API

## Important notes

### 1. Database Modifications:
- Added a foreign key between the `playlists` and `users` tables.
- Changed the constraints of the `cover`, `is_public`, and `description` columns in the `playlists` table to allow `NULL` values.

### 2. How to Test the APIs:
- Install new dependencies
  ```bash
   npm i

- Next, run the following command
   ```bash
   npm run playlist_test

- You should see a result like this:
![image](https://github.com/user-attachments/assets/9f0ad559-b8d1-4f36-9b58-337722a767a5)

- You could also test each API by thunder client. Here's an example for **_Create playlist_** API
  - Log in to obtain the `token`
  - Add `token` in the headers tab as shown in the image below:
![image](https://github.com/user-attachments/assets/59ab265f-1ac1-4249-972a-e862f84e51f0)
  - Add the request body:
  ![image](https://github.com/user-attachments/assets/b85e49be-869d-4ad8-a03f-fe47fd9e91fc)
  - Result:
  ![image](https://github.com/user-attachments/assets/837fa233-298a-4d05-8ad5-529513d05d0b)

If you wish to manually test any other APIs, you can easily do so by reviewing the code.
